admin_guide/out_of_resource_handling.adoc

```diff
@@ -0,0 +1,308 @@
+* `eviction-max-pod-grace-period` describes the maximum-allowed grace period (in
+seconds) to use when terminating pods in response to a soft eviction threshold
+being met.
+
+[[out-of-resource-hard-eviction-thresholds]]
+==== Hard Eviction Thresholds
+
+A hard eviction threshold has no grace period and, if observed, the node takes
+immediate action to reclaim the associated starved resource. If a hard eviction
+threshold is met, the node kills the pod immediately with no graceful
+termination.
+
+To configure hard eviction thresholds, the following flag is supported:
+
+* `eviction-hard` describes a set of eviction thresholds (for example,
+`memory.available<1Gi`) that, if met, triggers a pod eviction.
+
+[[out-of-resource-eviction-monitoring-interval]]
+=== Eviction Monitoring Interval
+
+The node evaluates eviction thresholds per its configured housekeeping interval.
+
+* `housekeeping-interval` is the interval between container housekeepings.
+
+[[out-of-resource-node-conditions]]
+=== Node Conditions
+
+The node will map one or more eviction signals to a corresponding node
+condition.
+
+If a hard eviction threshold is met, or a soft eviction threshold is met
+independent of its associated grace period, the node reports a condition
+indicating that the node is under pressure.
+
+The following node conditions are defined that correspond to the specified
+eviction signal.
+
+.Node Conditions Related to Low Resources
+[cols="2a,2a,8a",options="header"]
+|===
+
+|Node Condition |Eviction Signal |Description
+
+|`*MemoryPressure*`
+|`*memory.available*`
+|Available memory on the node has satisfied an eviction threshold.
+|===
+
+The node continues to report node status updates at the frequency specified by
+the `node-status-update-frequency` argument, which defaults to *10s*.
+
+[[out-of-resource-oscillation-of-node-conditions]]
+=== Oscillation of Node Conditions
+
+If a node is oscillating above and below a soft eviction threshold, but not
+exceeding its associated grace period, the corresponding node condition
+constantly oscillates between true and false, which can cause poor scheduling
+decisions.
+
+To protect against this oscillation, the following flag is defined to control
+how long the node must wait before transitioning out of a pressure condition.
+
+* `eviction-pressure-transition-period` is the duration for which the node has
+to wait before transitioning out of an eviction pressure condition.
+
+The node ensures that it has not observed an eviction threshold being met for
+the specified pressure condition for the period specified before toggling the
+condition back to *false*.
+
+[[out-of-resource-eviction-of-pods]]
+=== Eviction of Pods
+
+If an eviction threshold is met and the grace period is passed, the node
+initiates the process of evicting pods until it observes the signal going below
+its defined threshold.
+
+The node ranks pods for eviction:
+
+. By their quality of service, and
```

Comment:
Something we can link "quality of service" to explain the available classes? There's a "Quality of Service Classes" in `admin_guide/overcommit`.